### ISSUE TITLE
Petwalk for HA - Setup guide

### ISSUE THREAD

**Issue by Lars-Hiller** [p0l0/pypetwalk#1]
Dear Marco,

I was glad I found your repo to use the PetWalk Control info for Home Assistant.
Unfortunately, I was not able to get it running. Neither HACS nor HA AddOn Store accept your github repo link as valid. 

I downloaded the code and added it in the "custom_components" folder. But it was not recognized. Digging deeper in the requirements of custom components (in the [HACS documentation](https://hacs.xyz/docs/publish/integration/)), there seem to be files missing such as the manifest.json.

Long story short: Could you please give me a hint on how to get your integration running and how to set up the entities?

Thanks!
Best regards,
Lars

**Comment by p0l0** [comment]
Hi,

this is only the library which should be used by the [integration hapetwalk](https://github.com/p0l0/hapetwalk), but due to lack of time, I only created this library and the integration is still in development.

For my installation, i currently use [RESTful Sensors](https://www.home-assistant.io/integrations/sensor.rest/) as a workaround. All available options are documented [here](https://control.petwalk.solutions/doc/api/)

This is an example how i'm using it:

```yaml
switch:
  - platform: rest
    unique_id: petwalk_time
    name: Petwalk Time
    resource: http://<YOUR_PETWALK_IP>:8080/modes
    method: put
    headers:
      Accept: "application/json"
    body_on: '{"time": true}'
    body_off: '{"time": false}'
    is_on_template: "{{ value_json.time }}"
```

**Comment by Lars-Hiller** [comment]
Hi, 

thank you for the fast response and the details.
I will try this during the coming days. 

Thanks again for the work you put in!
Best regards,
Lars

**Comment by p0l0** [comment]
Here is the yaml code for all 4 modes:


```yaml
switch:
  - platform: rest
    unique_id: petwalk_time
    name: Petwalk Time
    resource: http://<YOUR_PETWALK_IP>:8080/modes
    method: put
    headers:
      Accept: "application/json"
    body_on: '{"time": true}'
    body_off: '{"time": false}'
    is_on_template: "{{ value_json.time }}"
 - platform: rest
    unique_id: petwalk_brightness_sensor
    name: Petwalk brightnessSensor
    resource: http://<YOUR_PETWALK_IP>:8080/modes
    method: put
    headers:
      Accept: "application/json"
    body_on: '{"brightnessSensor": true}'
    body_off: '{"brightnessSensor": false}'
    is_on_template: "{{ value_json.brightnessSensor }}"
  - platform: rest
    unique_id: petwalk_motion_in
    name: Petwalk Motion In
    resource: http://<YOUR_PETWALK_IP>:8080/modes
    method: put
    headers:
      Accept: "application/json"
    body_on: '{"motion_in": true}'
    body_off: '{"motion_in": false}'
    is_on_template: "{{ value_json.motion_in }}"
  - platform: rest
    unique_id: petwalk_motion_out
    name: Petwalk Motion Out
    resource: http://<YOUR_PETWALK_IP>:8080/modes
    method: put
    headers:
      Accept: "application/json"
    body_on: '{"motion_out": true}'
    body_off: '{"motion_out": false}'
    is_on_template: "{{ value_json.motion_out }}"
  - platform: rest
    unique_id: petwalk_rfid
    name: Petwalk Rfid
    resource: http://<YOUR_PETWALK_IP>:8080/modes
    method: put
    headers:
      Accept: "application/json"
    body_on: '{"rfid": true}'
    body_off: '{"rfid": false}'
    is_on_template: "{{ value_json.rfid }}"
```

**Comment by Lars-Hiller** [comment]
Hi,

thanks!
I figured those out myself. 
What I don't know how to do, are the door and the system state (neither read nor write). Have you set this up already?

Regards,
Lars

**Comment by p0l0** [comment]
I didn't implement it, as currently I only use the "time" mode as switch.

Do you want to implement it as a Switch (read/write) or as a Binary Sensor (read-only)?

**Comment by p0l0** [comment]
@Lars-Hiller i just finished the implementation and released [hapetwalk](https://github.com/p0l0/hapetwalk) as a custom component using HACS. Feel free to try it.

**Comment by Lars-Hiller** [comment]
Funktioniert hervorragend!
Besten Dank und Frohe Weihnachten!

Viele Grüße,
Lars
________________________________
Von: Marco ***@***.***>
Gesendet: Samstag, 23. Dezember 2023 15:33
An: p0l0/pypetwalk ***@***.***>
Cc: Lars-Hiller ***@***.***>; Mention ***@***.***>
Betreff: Re: [p0l0/pypetwalk] Petwalk for HA - Setup guide (Issue #1)


@Lars-Hiller<https://github.com/Lars-Hiller> i just finished the implementation and released hapetwalk<https://github.com/p0l0/hapetwalk> as a custom component using HACS. Feel free to try it.
